### ISSUE TITLE
make a easy  tutorial  how set it up for lemmy and reddit

### ISSUE THREAD

**Issue by BigDogg99** [everything-gripe/Infinity-For-Everything#1]
### Checklist

- [X] I have used the search function for [open](https://github.com/Docile-Alligator/Infinity-For-Reddit/issues) **and** [closed](https://github.com/Docile-Alligator/Infinity-For-Reddit/issues?q=is%3Aissue+is%3Aclosed) issues to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.
- [X] If the bug only occurs with a certain link, post, image..., I will include the URL.

### App version

latest

### Where did you get the app from

Other

### Android version

10

**Comment by theshatterstone** [comment]
@BigDogg99  I see that you've closed this issue.  May I ask why? I couldn't find a tutorial on using this app with reddit or lemmy. If you have, would you mind linking it here? Thanks.

**Comment by efraimbart** [comment]
> @BigDogg99 I see that you've closed this issue. May I ask why? I couldn't find a tutorial on using this app with reddit or lemmy. If you have, would you mind linking it here? Thanks.

There's no tutorial yet. If you have any specific questions feel free to ask and I'll try and answer.

If anyone would like to contribute to tutorial that would be helpful as well!

**Comment by theshatterstone** [comment]
@efraimbart I wanted to log in to my lemmy instance via infinity for everything alpha 2.1. When trying to log in, it asks me for a site and a client ID. For a site, I assume I just put in my Lemmy instance, for Client ID, I tried putting in my lemmy username, but that didn't work. What am I supposed to put in as a client ID? Thanks.

**Comment by efraimbart** [comment]
Everything login is currently a work in progress and is not yet available for Lemmy.

**Comment by theshatterstone** [comment]
@efraimbart So is there anything that can be done in the app currently? And do you have any timelines as to how long it would take to get Infinity working on Lemmy?

**Comment by efraimbart** [comment]
Of course! You can view posts, comments, communities and users on Lemmy using the Infinity for Everything.

Timelines are difficult, but progress is constantly being made on the backend with minimal app update requirements.

**Comment by theshatterstone** [comment]
@efraimbart Hi again, Just wanted to say that reddit connection seems to work in theory, and while the content of these subs would load when opening the subreddits themselves, after subscribing to some reddit subs, the home page is just loading forever. 

Also, I was able to see some lemmy communities on the popular section, but when searching for them, lemmy communities would not show up in search results on Infinity. In comparison, they showed up on Troddit or client.everything.gripe and I was even able to load the posts in the website, but not on Infinity.

**Comment by efraimbart** [comment]
There is some slowness when using Lemmy due to relying on the lemmy.world instance, but I would have to look into your specific issues.

Search is currently disabled on lemmy.world as well, see here https://www.everything.gripe/r/lemmy:support/comments/1634168/_/1388879/, but you can still navigate directly to a community via `Go to Subreddit` and entering something like `lemmy:android@lemdro.id`

**Comment by theshatterstone** [comment]
Alright, thank you very much! One last question: how does the reddit access work? Do you guys use web scraping or your own API? Thank you.

**Comment by efraimbart** [comment]
For now it's just without an api key

**Comment by theshatterstone** [comment]
So, Just leaving this here for anyone that needs it: 

## As of Alpha 2.1:



### To access Reddit communities:

Use the search function.

They will be displayed in the format:  `r/reddit.com:{name of subreddit}`



### To access Lemmy communities:

Use the `Go to Subreddit` function which is at the 3-dots menu -> `More Options` -> `Go to subreddit`

In there, type the name of your lemmy community in the format:
`r/lemmy:{name of community}@{name of instance}` e.g `lemmy:technology@lemmy.ml`

If your community is hosted at `lemmy.world`, you can skip the name of instance e.g `lemmy:asklemmy`

Don't worry if it doesn't appear in the dropdown, just make sure you've typed the name correctly and press `OK`

That will bring you to the lemmy page for that community. Click 'Subscribe' to, well, subscribe.



### Saving your Subscriptions:

Remember that accounts have not been implemented yet. If you want to save your list of subscribed communities, you can go to `Settings` -> `Advanced` -> `Backup Settings`

Note the fact that there is a backup password, which is `123321` by default. 

Within the backup archive, there is a directory named after the Infinity version e.g `5.4.2` 
In that folder, there is another folder called `database` and in it there is a file called `anonymous_subscribed_subreddits.json` where all of your subscriptions are. I'd suggest you save the archive as-is, rather than editing it.

**Comment by efraimbart** [comment]
Awesome, thanks @theshatterstone!

**Comment by theshatterstone** [comment]
@efraimbart If you're willing to create a wiki for the project, you can add this writeup to it. (But please add an attribution). This project shows a lot of promise, and I might switch to it, once accounts are added, and once it becomes a bit faster. In fact, I won't even wait for accounts as I've already got all my lemmy subscriptions on it. The only issue I currently have is the speed of fetching all of the posts. Currently, I use Liftoff for Lemmy and it fetches posts quite quickly for me. If Infinity for Everything starts fetching lemmy posts at those speeds, I'm willing to return to a familiar UI (I'm a former Infinity for Reddit user).

**Comment by efraimbart** [comment]
I'll keep that in mind. In the meantime, consider joining the [discord server](https://discord.gg/YN24JhbF4r)  for real-time discussion and updates.
